dts/bindings/serial/st,stm32-uart.yaml

```diff
@@ -2,21 +2,11 @@ description: STM32 UART
 
 compatible: "st,stm32-uart"
 
-include: uart-controller.yaml
+include: [uart-controller.yaml, pinctrl-device.yaml]
 
 properties:
     reg:
       required: true
 
     interrupts:
       required: true
-
-    pinctrl-0:
```

Comment:
To avoid frustration on build errors, the following should be added in all impacted bindings.
```
    pinctrl-names:
      required: true
```

Reply:
I initially planned to make `pinctrl-0` and `pinctrl-names` required, but then dropped the idea because Nordic needs to maintain backwards compatibility. Maybe nrf bindings can set required false?

Reply:
If possible, it would indeed make sense. But since we're doing this for interrupts and reg already, I guess we can live with the same on pinctrl-names...
But indeed, if it is not set as required in common binding, this is something we should take care when reviewing the related drivers changes.

Reply:
> Maybe nrf bindings can set required false?

You can't "downgrade" a `required: true` to a `required: false`:

> Note that weakening requirements by having required: false where the included file has required: true is an error. This is meant to keep the organization clean.

https://docs.zephyrproject.org/latest/guides/dts/bindings.html#include

Reply:
>  if it is not set as required in common binding, this is something we should take care when reviewing the related drivers changes.

so I think this is the way to go